File: Frontend/src/app/locacao/model/locacao.ts
import { Item } from "src/app/item/model/item";
import { Cliente } from "src/app/socio/model/cliente";

export interface Locacao {
  id: number;
  dtLocacao: Date;
  dtDevolucaoPrevista: Date;
  dtDevolucaoEfetiva: Date;
  valorCobrado: number;
  multaCobrada: number;
  item: Item;
  cliente: Cliente;
}

File: Backend/src/main/java/com/backend/model/Locacao.java
package com.backend.model;

import java.time.LocalDate;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "locacao")
public class Locacao {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column
    private LocalDate dtLocacao;
    
    @Column
    private LocalDate dtDevolucaoPrevista;
    
    @Column
    private LocalDate dtDevolucaoEfetiva;
    
    @Column
    private Double valorCobrado;
    
    @Column
    private Double multaCobrada;

    @ManyToOne
    @JoinColumn(name="id_cliente", referencedColumnName = "id")
    private Cliente cliente;

    @ManyToOne
    @JoinColumn(name="id_item", referencedColumnName = "id")
    private Item item;


    
}
